community/kernel/src/main/java/org/neo4j/kernel/impl/util/collection/TimedRepository.java

```diff
@@ -203,6 +193,12 @@ public void release( KEY key )
         return repo.keySet();
     }
 
+    public VALUE getValue( KEY key )
+    {
+        Entry entry = repo.get( key );
+        return entry == null ? null : entry.value;
+    }
+
```

Comment:
Instead of exposing this unsafe method, you can make `ConversationManager.createConversationStore` protected, and let the `MasterImplConversationStopIT`override it with a version where the factory and the reaper collaborate on maintaining a `ConcurrentHashMap` where the `RequestContexts` are mapped to `Conversations`.